### DISCUSSION TITLE
Document XMAKE_STATS

### DISCUSSION THREAD

**Discussion by hg-zt** [xmake-io/xmake#1802] [General]
xmake collects stats by cloning an empty repo

https://github.com/xmake-io/xmake/blob/79acd86359c71f749b6990b24335bfe2c338f95b/xmake/actions/build/statistics.lua

but this behavior is not documented anywhere. It is enabled by default and could be disabled by setting an environment variable `XMAKE_STATS=false` which is not documented either (I found it by looking into the source code).

I understand that the current implementation (cloning an empty github repo) could hardly expose any information or do harm to users, but I simply do not want any network traffic that I do not need, especially that I'm not aware of.

My suggestion is that `xmake.io` should let users know what happens and how to disable it because it is not expected by most users.
Another thought is that xrepo is a better place for collecting stats. No one would be surprised that npmjs.com, crates.io(rust) or conan.io show download/search/query stats.

I do think xmake is a good and modern alternative tool for cmake and hope it become popular among c/c++ communities, while collecting information is rather a sensitive and controversial behavior that might do more harm than good in the future.

**Comment by waruqi** [comment]
There is currently no better way to count the usage of xmake. Statistics using xrepo are not accurate, and many users do not use packages. In addition, all dependent packages currently installed by xrepo are directly accessed original URLs, and I cannot count their download information.

Currently, I only use github traffics to count them, which will not reveal user privacy. In addition, a clone of an empty project uses very little traffic.

Of course, users have the right to disable them, so I provide XMAKE_STATS to disable this behavior.

I will update the document in the next few days and clearly explain its purpose and how to disable it.

The purpose of adding this statistic is to let me know the approximate user growth of xmake so that I can have enough confidence to continue to maintain it.

In the future, if there is a better way to count it, or when there are enough users, I feel that it is no longer needed. I will remove it.

**Comment by waruqi** [comment]
I have added it in documents. https://xmake.io/#/guide/configuration?id=xmake_stats
